### PR TITLE
Rewrite and Optimize Script

### PR DESCRIPTION
**Pull request by MichaelCoding25**
Disclaimer: This does not change how the script behaves for the player!

- Update __resource to fxmanifest
- Refactor script
- Lower MS usage and implement garbage collection to combat memory leak
- Expand config and add more customizations
- Update natives to be more recent and use correct names

### CODE REVIEW

**Comment by MichaelCoding25** [comment]
I have tested this and it works

File: script_armory/client.lua [hunk 1]
```diff
@@ -8,111 +8,107 @@
...
-                                weaponComponent(comp[1], comp[2])
-                               
+        Citizen.Wait(200)
+
+        local ped = PlayerPedId()
+        local coords = GetEntityCoords(ped)
+
+        local lowestDist = Config.Marker.DrawDistance
+        for _,armory in pairs(Config.Armories) do
+            local distance = GetDistanceBetweenCoords(coords, armory.x, armory.y, armory.z, true)
```
Comment (BaziForYou) [inline script_armory/client.lua]: it's better to use this for optimizing
```lua
local distance = #(coords - vector3(armory.x, armory.y, armory.z))
```

File: script_armory/client.lua [hunk 2]
```diff
@@ -8,111 +8,107 @@
...
 
-  
-  function giveWeapon(hash)
-    GiveWeaponToPed(GetPlayerPed(-1), GetHashKey(hash), 100, false, true)
-  end
-  
-  function weaponComponent(weapon, comp)
-    if HasPedGotWeapon(GetPlayerPed(-1), GetHashKey(weapon), false) then
-        GiveWeaponComponentToPed(GetPlayerPed(-1), GetHashKey(weapon), GetHashKey(comp))
+function weaponComponent(weapon, comp)
```
Comment (BaziForYou) [inline script_armory/client.lua]: better to get ped from outside and don't call to get player for optimizing
```lua
function weaponComponent(ped, weapon, comp)
```

File: script_armory/client.lua [hunk 3]
```diff
@@ -8,111 +8,107 @@
...
+    local ped = PlayerPedId()
```
Comment (BaziForYou) [inline script_armory/client.lua]: remove this you got this from outside

File: script_armory/client.lua [hunk 4]
```diff
@@ -8,111 +8,107 @@
...
+    AddTextComponentSubstringPlayerName(text)
+    EndTextCommandThefeedPostTicker(false, false)
 end
 
-function ShowNotification(text)
-    SetNotificationTextEntry("STRING")
-    AddTextComponentString(text)
-    DrawNotification(false, false)
+
+function giveWeapon(hash)
```
Comment (BaziForYou) [inline script_armory/client.lua]: better to get ped from outside and don't call to get player for optimizing
```lua
function giveWeapon(ped, hash)
```

File: script_armory/client.lua [hunk 5]
```diff
@@ -8,111 +8,107 @@
...
+    GiveWeaponToPed(PlayerPedId(), hash, 100, false, false)
```
Comment (BaziForYou) [inline script_armory/client.lua]: better to get ped from outside and don't call to get player for optimizing
```lua
GiveWeaponToPed(ped, hash, 100, false, false)
```

File: script_armory/client.lua [hunk 6]
```diff
@@ -8,111 +8,107 @@
...
+                    local ped = PlayerPedId()
 
+                    RemoveAllPedWeapons(ped, true)
 
+                    for _,weapon in ipairs(Config.Weapons) do
+                        giveWeapon(weapon)
+                    end
 
+                    for _,comp in ipairs(Config.Components) do
+                        weaponComponent(comp.weapon, comp.component)
```
Comment (BaziForYou) [inline script_armory/client.lua]: better to send ped from here and don't call to get player inside of  function for optimizing
```lua
weaponComponent(ped, comp.weapon, comp.component)
```

File: script_armory/client.lua [hunk 7]
```diff
@@ -8,111 +8,107 @@
...
+            if closestArmory.isInMarker then
+                ShowAlert(Config.Message)
 
+                if IsControlJustPressed(0, Config.Keybind) then
+                    local ped = PlayerPedId()
 
+                    RemoveAllPedWeapons(ped, true)
 
+                    for _,weapon in ipairs(Config.Weapons) do
+                        giveWeapon(weapon)
```
Comment (BaziForYou) [inline script_armory/client.lua]: better to send ped from here and don't call to get player inside of  function for optimizing
```lua
giveWeapon(ped, weapon)
```

File: script_armory/client.lua [hunk 8]
```diff
@@ -8,111 +8,107 @@
...
+    end
+end)
 
-                              end
-                        if Config.armor then
-                              AddArmourToPed(GetPlayerPed(-1), 100)
-                        end
+Citizen.CreateThread(function()
+    while true do
+        Citizen.Wait(0)
```
Comment (BaziForYou) [inline script_armory/client.lua:51]: better to like this
```lua
local LetSleep = true
```

File: script_armory/client.lua [hunk 9]
```diff
@@ -8,111 +8,107 @@
...
 
+        if closestArmory.loc ~= {} then
+            local markerScale = Config.Marker.Scale
+            local markerColor = Config.Marker.Color
```
Comment (BaziForYou) [inline script_armory/client.lua:59]:
```lua
LetSleep = false
```

File: script_armory/client.lua [hunk 10]
```diff
@@ -8,111 +8,107 @@
...
+                    end
 
+                    if Config.Armor then
+                        AddArmourToPed(ped, 100)
+                    end
 
+                    ShowNotification("~g~Police loadout has been loaded.")
+                end
+            end
+        end
```
Comment (BaziForYou) [inline script_armory/client.lua:85]: and add 

```lua
if LetSleep then
	Citizen.Wait(1000)
end
```

File: script_armory/client.lua [hunk 11]
```diff
@@ -8,111 +8,107 @@
...
-                              for _,weapon in ipairs(Config.Weapons) do
-                                giveWeapon(weapon)
-
-
-                              end
-
-                              for _,comp in ipairs(Config.Components) do
-                                weaponComponent(comp[1], comp[2])
-                               
+        Citizen.Wait(200)
```
Comment (BaziForYou) [inline script_armory/client.lua]: better to sleep more like one sec
```lua
Citizen.Wait(1000)
```

**Pull request closed without merging**